nano/node/network.cpp

```diff
@@ -891,32 +891,35 @@ void nano::tcp_message_manager::put_message (nano::tcp_message_item const & item
 {
 	{
 		nano::unique_lock<std::mutex> lock (mutex);
-		while (entries.size () > max_entries && !stopped)
+		while (entries.size () >= max_entries && !stopped)
 		{
 			condition.wait (lock);
 		}
 		entries.push_back (item_a);
 	}
-	condition.notify_all ();
+	condition.notify_one ();
```

Comment:
I don't think we want to notify_one here, since multiple threads could be waiting on the queue to be less than max_entries, it might wake up one of those threads instead of the consumer.

Reply:
oh good spot, I missed that. I wonder if the producer/consumer should have different condition variables so the correct thread can be woken up instead of all.

Reply:
Requested from @guilhermelawless to make necessary changes